x-pack/plugins/encrypted_saved_objects/index.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { Request } from 'hapi';
+import { Root } from 'joi';
+import { Legacy } from 'kibana';
+
+import { EncryptedSavedObjectsClientWrapper } from './server/lib/encrypted_saved_objects_client_wrapper';
+import {
+  EncryptedSavedObjectsService,
+  EncryptedSavedObjectTypeRegistration,
+} from './server/lib/encrypted_saved_objects_service';
+import { createLogger } from './server/lib/logger';
+
+export const encryptedSavedObjects = (kibana: any) =>
+  new kibana.Plugin({
+    id: 'encrypted_saved_objects',
+    configPrefix: 'xpack.encrypted_saved_objects',
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+
+    config(Joi: Root) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        encryptionKey: Joi.string().min(32),
+        audit: Joi.object({ enabled: Joi.boolean().default(false) }),
+      }).default();
+    },
+
+    async init(server: Legacy.Server) {
+      const config = server.config();
+      let encryptionKey = config.get<string>('xpack.encrypted_saved_objects.encryptionKey');
+      if (encryptionKey == null) {
+        server.log(
+          ['warning', 'encrypted-saved-objects'],
+          'Generating a random key for xpack.encrypted_saved_objects.encryptionKey. To be able ' +
+            'to decrypt encrypted saved objects attributes after restart, please set ' +
+            'xpack.encrypted_saved_objects.encryptionKey in kibana.yml'
+        );
+
+        encryptionKey = crypto.randomBytes(16).toString('hex');
+      }
+
+      const service = Object.freeze(
+        new EncryptedSavedObjectsService(
+          encryptionKey,
+          server.savedObjects.types,
+          createLogger(server)
+        )
+      );
+
+      server.expose('registerType', (typeRegistration: EncryptedSavedObjectTypeRegistration) =>
+        service.registerType(typeRegistration)
+      );
+
+      // This "hidden" option is used to tell `EncryptedSavedObjectsClientWrapper` to not strip
```

Comment:
__note:__ that's the least hackish way I found to do what I want :) Do you have any better ideas?

Reply:
I see. I'm wondering if we need to have two different methods then? One of which will be `request scoped` and another one isn't? I was thinking about a user within a space that is creating *some-saved-object-with-encrypted-attributes* and then editing them and hence it'd make sense to involve space and security for such requests. But not sure if it's a valid use case.

Reply:
Summary of out-of-github discussion: since we know only one potential consumer of this API (Alerting) and this consumer will need to get decrypted values without request we'll keep only `getDecryptedAsInternalUser` that will be accessing underlying repository directly for the time being.